### PR TITLE
DOC: Add Sphinx documentation

### PR DESCRIPTION
**Pull request by MartinThoma**

### CODE REVIEW

**Comment by MartinThoma** [comment]
@mstamy2 / @MasterOdin Could you please give me the necessary permissions to add webhooks?

I would like to configure automatic documentation updates with readthedocs: https://docs.readthedocs.io/en/latest/integrations.html

I'm uncertain where the documentation on https://pythonhosted.org/PyPDF2/ came from, but I feel like we can do better (especially adding user guides)

**Comment by MasterOdin** [comment]
The docs at pythonhosted.org are also probably not updatable anymore as I think PyPI has since deprecated / removed the ability to update them, with the suggestion to use RTD or gh-pages.

Similar to the other issue, I think only @mstamy2 has power to do those things and I don't think he can delegate that to someone else. To have multiple "owners" and people who can change settings, would have to migrate the repo to an organization.

File: docs/modules/REAMDE.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Module documentation
```
Comment (MasterOdin) [inline docs/modules/REAMDE.md:1]: Should this be named `README`, not `REAMDE`?

**Review by MasterOdin — CHANGES_REQUESTED** [review]

**Comment by mstamy2** [comment]
@MartinThoma I can transfer the owner role to you which should give you all the powers you need - definitely think you've earned it (:

Can still always go the route of transferring to an org after the initial clean-up if you wanted

The original ReST files for the Sphinx docs might be lost (yep - definitely should have been part of the repo), but looks like you've replaced them all. Maybe just the `about` page and that `pdfcat` script tutorial are missing, but they're outdated anyway

**Comment by MartinThoma** [comment]
That sounds awesome! I would understand if you would want to wait a couple of months with transferring ownership. In the end, you don't know me and I first need to earn the trust.

**Pull request merged**